### ISSUE TITLE
`Audacity` playback cursor doesn't get updated when running under Wayland

### ISSUE THREAD

**Issue by MikiVanousek** [NixOS/nixpkgs#238910]
### Describe the bug
When I start playback in audacity, the cursor UI will not get updated, until something else triggers a UI update. This can be anything from, hovering over a button to clicking an empty part of the window. See [the video](https://streamable.com/gjwf1k) I made, when you see the playback cursor jump ahead repeatedly at the end, it is because I am clicking.

### Additional context
I found the problem and the workaround [here,](https://forum.audacityteam.org/t/cursor-not-moving-during-playback/56035/18) and running audacity under X11 did solve the issue.

```console
Exec=env GDK_BACKEND=x11 audacity
```

I also see these errors in the shell, but they are still there if I use the workaround:
```console
(process:472241): Gdk-CRITICAL **: 10:04:43.477: gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed

(process:472241): Gdk-CRITICAL **: 10:04:43.477: gdk_window_get_display: assertion 'GDK_IS_WINDOW (window)' failed

(process:472241): Gdk-CRITICAL **: 10:04:43.477: gdk_cursor_new_from_pixbuf: assertion 'GDK_IS_DISPLAY (display)' failed
```

### Notify maintainers
@lheckemann 
@veprbl 
@wegank 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.34, NixOS, 23.05 (Stoat), 23.05.885.bb8b5735d6f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(miki): `""`
 - channels(root): `"nixos-23.05, nixos-unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by wegank** [comment]
See audacity/audacity#459, I'm not sure we can do anything downstream here.

**Comment by MikiVanousek** [comment]
Is there a way to change the derivation, in a way it will run audacity under X11 (using the workaround as default)?
```
Exec=env GDK_BACKEND=x11 audacity
```

**Comment by MikiVanousek** [comment]
But that would only fix the issue for me, instead for anyone who installs the package, right?

**Comment by wegank** [comment]
True. The workaround might not be generally applicable, for example, it would break Audacity on a pure Wayland configuration (i.e. with XWayland disabled).

**Comment by praduca** [comment]
using `environment.shellAliases` work from command line, but not from gnome (by clicking the icon). I can live with it, but if there is a way that configure this for both? Or i am doing something wrong here?

**Comment by AndrewKvalheim** [comment]
You can add something like these—

https://github.com/NixOS/nixpkgs/blob/e99719836d6a3aff0cf4b601339427e2cd91404c/pkgs/by-name/gm/gmrun/package.nix#L29-L32

https://github.com/NixOS/nixpkgs/blob/e99719836d6a3aff0cf4b601339427e2cd91404c/pkgs/applications/graphics/viewnior/default.nix#L49-L53

—with an override:

```nix
# Work around NixOS/nixpkgs#238910
audacity = pkgs.audacity.overrideAttrs (a: {
  preFixup = a.preFixup or "" + ''
    gappsWrapperArgs+=(--set-default GDK_BACKEND x11)
  '';
});
```
